[docs] Add missing pickers migration docs (#7000)
mui/mui-x: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md

These changes were done for consistency, improve stability and make room for new features.
Below are described the steps you need to make to migrate from v5 to v6.

### Drop `clock` in desktop mode

> We have dropped the clock picker behavior in desktop mode for `DateTimePicker` and `TimePicker` components.

Author: @joserodolfofreitas or any1 else—feel free to suggest improvements on this or the below sections. 🙏

Reviewer: In desktop mode, the `DateTimePicker` and `TimePicker` components will not display the clock.

Author: Updated 😉

This is the first step towards moving to a [better implementation](https://github.com/mui/mui-x/issues/4483).
The behavior on mobile mode is still the same.
If you were relying on Clock Picker in desktop mode for tests—make sure to check [testing caveats](/x/react-date-pickers/getting-started/#testing-caveats) to choose the best replacement for it.

> ### Drop direct `@date-io` adapters support

Reviewer: It sounds like we are dropping `@date-io`

```suggestion
### Extended `@date-io` adapters
```

> In order to support new Date and Time pickers with [fields components](/x/react-date-pickers/fields/) we had to augment `@date-io` adapters.

Reviewer: In v5, it was possible to import adapters either from `@date-io` or `@mui/x-date-pickers` which were the same.
In v6, the adapters are extended by `@mui/x-date-pickers` to support field components. 
Which means adapters can not be imported from `@date-io` anymore. They might be imported from `@mui/x-date-pickers` or `@mui/x-date-pickers-pro`. Otherwise, some methods will be missing.

Author: Updated with some tweaks to the whole paragraph. 👌

If you were using adapter from `@date-io`—you will need to update the import.

> If we do not have a specific augmented adapter you were using—there probably was a reason for it, but you can raise an issue expressing interest in it.

Reviewer: ```suggestion
If you do not find the adapter you were using—there probably was a reason for it, but you can raise an issue expressing interest in it.
```

   <DatePicker
  -  toolbarTitle="Title"
  +  localeText={{ toolbarTitle: "Title" }}
   />
  ```

- The toolbar related translation keys have been renamed to better fit their usage:

  ```diff
   // Same on all other pickers
   <DatePicker
    localeText={{
  -    datePickerDefaultToolbarTitle: 'Date Picker',
  +    datePickerToolbarTitle: 'Date Picker',

  -    timePickerDefaultToolbarTitle: 'Time Picker',
  +    timePickerToolbarTitle: 'Time Picker',

  -    dateTimePickerDefaultToolbarTitle: 'Date Time Picker',
  +    dateTimePickerToolbarTitle: 'Date Time Picker',

  -    dateRangePickerDefaultToolbarTitle: 'Date Range Picker',
  +    dateRangePickerToolbarTitle: 'Date Range Picker',
    }}
   />
  ```

- The `onChange` / `openView` props on the toolbar have been renamed to `onViewChange` / `view`

### Replace `tabs` props

- The `hideTabs`, `dateRangeIcon`, and `timeIcon` props have been moved to `tabs` component props:

  ```diff
   // Same on all other Date Time picker variations
   <DateTimePicker
  -  hideTabs={false}
  -  dateRangeIcon={<LightModeIcon />}
  -  timeIcon={<AcUnitIcon />}
  +  componentsProps={{
  +    tabs: {
  +      hidden: false,

>   +      dateRangeIcon: <LightModeIcon />,

Author: This icon name is a bit strange to me.
What do you think about renaming it to `dateIcon` to be more in line with `timeIcon` @alexfauquette @flaviendelangle ? 🤔

Reviewer 2: The name of the prop come from the name of the default icon
But I do agree that in the context of the picker it's very confusing

Reviewer: +1 for changing to `dateIcon` which would feel more natural. Better having a meaning full prop naming than to match the default icon name

Author: Thank you for the comments.
I've [created a PR](https://github.com/mui/mui-x/pull/7024) to handle it.
Just waiting for this one to be merged firstly. 👌

  +      timeIcon: <AcUnitIcon />,
  +    }
  +  }}
   />
  ```

> - The `onChange` prop on `DateTimePickerTabs` component has been renamed to `onViewChange` to better fit its usage:

Reviewer 2: The actual breaking change would be if user try to set a custom tab component
I would rephrase it the opposite way

Something like

````suggestion
- The `onChange` prop of the `tabs` component slot has been renamed `onViewChange`

```diff
const CustomTabComponent = props => (
  <div>
-  <button onClick={() => props.onChange('day')}>Show day view</button>
+  <button onClick={() => props.onViewChange('day')}>Show day view</button>
  </div>
)

<DateTimePicker
  components={{
    tabs: CustomTabComponent
  }}
/>
```
````

Author: Or they have used our publicly exported `DateTimePickerTabs` to build something custom. 🤔 
I'd say that such changes we could include both the current diff and my suggestion—possibly leaving the text the same..? 🤔

Reviewer 2: You are right, we could do both

Author: Updated 😉
